server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/DeadCompactionDetector.java

```diff
@@ -75,6 +82,17 @@ private void detectDeadCompactions() {
       tabletCompactions.forEach((ecid, extent) -> log.trace("Saw {} for {}", ecid, extent));
     }
 
+    // Remove from the dead map any compactions that the Tablet's
+    // do not think are running any more.
+    this.deadCompactions.keySet().forEach(eci -> {
+      if (!tabletCompactions.containsKey(eci)) {
+        if (this.deadCompactions.remove(eci) != null)
+          log.trace(
+              "Removed {} from the dead compaction map, no tablet thinks this compaction is running",
+              eci);
+      }
+    });
```

Comment:
I think the following does the same thing.

```suggestion
    this.deadCompactions.keySet().retailAll(tabletCompactions.keySet());
```

Reply:
thanks. That's much simpler.

Reply:
I applied this change in 95e8fec149722ac3f205734103223e3f42b33438